internal/driver/macmapper_test.go

```diff
@@ -178,3 +182,233 @@ func TestMatchEndpointRefAddressToMAC(t *testing.T) {
+			mockSecretProvider := &mocks.SecretProvider{}
+			mockLoggingClient := logger.MockLogger{}
+
+			if len(test.currentMap) == 0 {
+				return
+			}
+
+			for secretPath := range test.currentMap {
+				if strings.ToLower(secretPath) != noAuthSecretPath {
+					mockSecretProvider.On("GetSecret", secretPath, UsernameKey, PasswordKey, AuthModeKey).
+						Return(nil, nil)
+				}
+			}
+
+			mockService.On("GetSecretProvider").
+				Return(mockSecretProvider)
+			mockService.On("GetLoggingClient").Return(mockLoggingClient)
+			driver.macAddressMapper.UpdateMappings(test.currentMap)
+
+			if test.name == "duplicate macs" {
+				ex1 := reflect.DeepEqual(test.expected, driver.macAddressMapper.credsMap)
+				ex2 := reflect.DeepEqual(test.alternateExpected, driver.macAddressMapper.credsMap)
+				assert.Equal(t, true, (ex1 || ex2))
+				return
+			}
+			assert.Equal(t, test.expected, driver.macAddressMapper.credsMap)
+		})
+	}
+}
+
+// TestMACAddressMapperListMACAddresses verifies the correct list of MAC addresses are returned that have credentials attached to them.
+func TestMACAddressMapperListMACAddresses(t *testing.T) {
+	tests := []struct {
+		name     string
+		credMap  map[string]string
+		expected []string
+	}{
+		{
+			name: "single entry",
+			credMap: map[string]string{
+				"AA:BB:CC:DD:EE:FF": "creds1",
+			},
+			expected: []string{
+				"AA:BB:CC:DD:EE:FF",
+			},
+		},
+		{
+			name: "double entry same path",
+			credMap: map[string]string{
+				"AA:BB:CC:DD:EE:FF": "creds1",
+				"11:22:33:44:55:66": "creds1",
+			},
+			expected: []string{
+				"AA:BB:CC:DD:EE:FF",
+				"11:22:33:44:55:66",
+			},
+		},
+		{
+			name: "double entry different path",
+			credMap: map[string]string{
+				"AA:BB:CC:DD:EE:FF": "creds1",
+				"11:22:33:44:55:66": "creds2",
+			},
+			expected: []string{
+				"AA:BB:CC:DD:EE:FF",
+				"11:22:33:44:55:66",
+			},
+		},
+		{
+			name: "many entries different paths",
+			credMap: map[string]string{
+				"AA:BB:CC:DD:EE:FF": "creds1",
+				"11:22:33:44:55:66": "creds2",
+				"AA:BB:CC:FF:EE:DD": "creds1",
+				"FF:EE:DD:CC:BB:AA": "creds1",
+				"66:55:44:33:22:11": "creds2",
+			},
+			expected: []string{
+				"AA:BB:CC:DD:EE:FF",
+				"11:22:33:44:55:66",
+				"AA:BB:CC:FF:EE:DD",
+				"FF:EE:DD:CC:BB:AA",
+				"66:55:44:33:22:11",
+			},
+		},
+		{
+			name:     "no entries",
+			credMap:  map[string]string{},
+			expected: []string{},
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			m := &MACAddressMapper{
+				credsMu:  sync.RWMutex{},
+				credsMap: test.credMap,
+			}
+			actual := m.ListMACAddresses()
+			assert.ElementsMatch(t, test.expected, actual)
+		})
+	}
+}
+
+// TestTryGetSecretPathForMACAddress verifies the correct secret path is returned for a given mac address.
+func TestTryGetSecretPathForMACAddress(t *testing.T) {
+
+	tests := []struct {
+		mac      string // input mac address
+		expected string
+	}{
+		{
+			mac:      "aa:bb:cc:dd:ee:ff",
```

Comment:
i think these test cases do need a name its confusing what the different macs mean